pg_shard.c

```diff
@@ -393,7 +393,9 @@ ErrorIfQueryNotSupported(Query *queryTree)
 	bool hasValuesScan = false;
 	uint32 queryTableCount = 0;
 	bool hasNonConstTargetEntryExprs = false;
+	bool hasNonConstQualExprs = false;
 	bool specifiesPartitionValue = false;
+	Node *whereClause = NULL;
```

Comment:
Instead of `whereClause`, I think you just need to declare `FromExpr *joinTree;` inside the `if` block.